src/cpu/x64/matmul/amx_blocking_heuristics.cpp

```diff
@@ -0,0 +1,1053 @@
+    if (bgmmc.K < bgmmc.wei_k_blk
+            || bgmmc.K % data_type_vnni_granularity(bgmmc.wei_dt) != 0) {
+        return false;
+    }
+
+    bool a_dt_ok
+            = one_of(bgmmc.orig_src_dt, dnnl_s8, dnnl_u8, dnnl_bf16, dnnl_f16);
+    bool b_dt_ok
+            = one_of(bgmmc.orig_wei_dt, dnnl_s8, dnnl_u8, dnnl_bf16, dnnl_f16);
+
+    bool a_tag_ok = bgmmc.src_tag == dnnl_format_tag_any
+            || bm_conf_utils.check_is_plain(bgmmc.src_tag);
+    bool b_tag_ok = bm_conf_utils.is_any_B_layout()
+            || bm_conf_utils.check_b_layout_blocked_32_by_n(bgmmc.wei_tag);
+
+    bool has_zp = bgmmc.src_zp_type != brgemm_broadcast_t::none
+            || bgmmc.wei_zp_type != brgemm_broadcast_t::none
+            || bgmmc.dst_zp_type != brgemm_broadcast_t::none;
+
+    return bgmmc.orig_src_dt == bgmmc.src_dt
+            && bgmmc.orig_wei_dt == bgmmc.wei_dt && bgmmc.is_amx
+            && !bgmmc.is_runtime_N && !bgmmc.is_runtime_M && a_dt_ok && a_tag_ok
+            && (bgmmc.reduce_kind == matmul_reduce_kind::undef) && b_tag_ok
+            && b_dt_ok && !has_zp;
+}
+
+bool matmul_amx_blocking_params_macro_t::divs_are_acceptable() {
+    bool unacceptable_m_div = m_per_thread < min_m_dim && nthr_m_ > 1;
+    bool unacceptable_k_div = k_per_thread < min_k_dim && nthr_k_ > 1;
+    bool unacceptable_n_div;
+    if (nthr_k_ == 1 && k_per_thread < k_threshold_write_bound_layer) {
+        // The layer is write bound (small K) and no reduction (C becomes non consecutive)
+        unacceptable_n_div
+                = n_per_thread < min_n_dim_write_bound_layer && nthr_n_ > 1;
+    } else {
+        unacceptable_n_div = n_per_thread < min_n_dim && nthr_n_ > 1;
+    }
+
+    bool unacceptable_b_div = nthr_b_ > (size_t)batch;
+
+    return !unacceptable_m_div && !unacceptable_k_div && !unacceptable_n_div
+            && !unacceptable_b_div;
+}
+
+size_t determine_tmul_size(size_t num_elements, int full_tile_size) {
+    size_t tmul_tiles = div_up(num_elements, full_tile_size);
+    size_t tmul_size = div_up(num_elements, tmul_tiles);
+    return tmul_size;
+}
+
+bool matmul_amx_blocking_params_macro_t::find_best_blocking(
+        const brgemm_matmul_conf_t &bgmmc,
+        const brgemm_matmul_conf_utils_t &bm_conf_utils,
+        matmul_amx_blocking_params_macro_t &best_blocking) {
+
+    if (!matmul_amx_blocking_params_macro_t::is_supported(
+                bgmmc, bm_conf_utils)) {
+        return false;
+    }
+
+    best_blocking = matmul_amx_blocking_params_micro_t(bgmmc);
+
+    matmul_amx_blocking_params_macro_t current_blocking(bgmmc);
+    assert(bgmmc.tr_a_dt_sz == bgmmc.tr_b_dt_sz);
+    current_blocking.gemm_dt_sz = bgmmc.tr_a_dt_sz;
+
+    for (size_t nthr_to_check = bgmmc.nthr; nthr_to_check > 0;
+            nthr_to_check--) {
+        current_blocking.nthr_ = nthr_to_check;
+
+        for (int b_div = 1; b_div <= current_blocking.nthr_; ++b_div) {
+            if (current_blocking.nthr_ % b_div == 0) {
```

Comment:
Would better for code alignment:
```suggestion
            if (current_blocking.nthr_ % b_div != 0) continue;
```

Reply:
fixed